acceptance: Move HostConfig.NetworkMode setting into createContainer

bdarnell opened the pull request:
This puts the containers used for cross-language testing on the same
network as the nodes and gets the tests passing reliably for me.
(I don't understand why the tests were passing some of the time before).

Removed redundant alias specification and restricted the DNS search path
to avoid querying outside networks.

Fixes #4894

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4953)

<!-- Reviewable:end -->

bdarnell committed:
acceptance: Move HostConfig.NetworkMode setting into createContainer

This puts the containers used for cross-language testing on the same
network as the nodes and gets the tests passing reliably for me.
(I don't understand why the tests were passing some of the time before).

Removed redundant alias specification and restricted the DNS search path
to avoid querying outside networks.

Fixes #4894

tamird commented:
LGTM

petermattis commented on acceptance/cluster/localcluster.go:
```diff
@@ -336,7 +335,6 @@ func (l *LocalCluster) createRoach(node *testNode, vols *Container, cmd ...strin
 
 	hostConfig := container.HostConfig{
 		PublishAllPorts: true,
-		NetworkMode:     container.NetworkMode(l.networkID),
```
How is `l.networkID` being used after this removal?

tamird replied:
It's in createContainer (as mentioned in the commit message)

On Mon, Mar 7, 2016 at 4:31 PM, Peter Mattis <notifications@github.com>
wrote:

> In acceptance/cluster/localcluster.go
> <https://github.com/cockroachdb/cockroach/pull/4953#discussion_r55275608>:
>
> > @@ -336,7 +335,6 @@ func (l *LocalCluster) createRoach(node *testNode, vols *Container, cmd ...strin
> >
> >  	hostConfig := container.HostConfig{
> >  		PublishAllPorts: true,
> > -		NetworkMode:     container.NetworkMode(l.networkID),
>
> How is l.networkID being used after this removal?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/4953/files#r55275608>.
>

petermattis replied:
Ah, just missed that (I read the commit message, but wasn't reading the code properly).

petermattis commented:
LGTM. Did you test this with a fouled up "domain" specification in `/etc/resolv.conf` like I had?

bdarnell committed:
acceptance: default IP for docker is localhost, not 0.0.0.0

bdarnell committed:
acceptance: remove variables from cross-language configs

The containers address each other from within the docker network,
not the host machine.

bdarnell commented:
I've added two new commits that seem to have fixed the issues that were blocking this PR (but I've had to rebuild several times to get green, for failures that are not obviously related). I don't quite understand how things were working before; the old code just looks wrong to me.

tamird commented:
Reviewed 2 of 2 files at r1, 2 of 2 files at r2, 2 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4953)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4953#-:-KCTu1SuC1xtQE1yrK2_:1417088721)*
<!-- Sent from Reviewable.io -->

petermattis commented:
LGTM
